### PR TITLE
Populate original genesis ledger balances during archive migration

### PR DESCRIPTION
**Pull request by joaosreis**
This pull request adds logic to populate the original genesis ledger balances during berkeley archive migration. It creates the genesis ledger, retrieves the account IDs from the ledger and respective ledger indexes, and adds entries to the `accounts_accessed` table of the migrated database for each genesis ledger account. Similarly to the block migration process, it migrates accounts in a sorted way (by ledger index) so that if the process is interrupted it can be resumed from the last ledger index.
It was tested by running a migration using the current UMT network original genesis ledger and running a script that checks that every account on the runtime config genesis ledger exists on the migrated database and that the data (balance, timing info, delegation, ...) matches.
I also observed that it's possible to run the migration app against an already migrated DB (without genesis balances) and it will migrate the genesis balances without any issue or additional requirement.

### CODE REVIEW

File: check_db_genesis_balances.py
Comment (dkijania) [inline check_db_genesis_balances.py:1]: I think we should move this script to scripts/archive maybe?

File: check_db_genesis_balances.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+host = os.environ.get("DB_HOST", "localhost")
+port = os.environ.get("DB_PORT", "5432")
+config_file = os.environ.get("CONFIG_FILE")
+
+
+parser = argparse.ArgumentParser(
+    description="Check that the genesis balances in the JSON file match the database."
+)
+parser.add_argument(
+    "--config-file",
```
Comment (dkijania) [inline check_db_genesis_balances.py:21]: NIT: config file is to open name. Do you mean ledger json or replayer config file?

File: check_db_genesis_balances.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        ):
+            messages.append(f"nonce: {account['nonce']} != {result['nonce']}")
+
+    if len(messages) != 0:
+        all_accounts_match = False
+        print(f"Account with pk '{account['pk']}' does not match the SQL query result.")
+        for message in messages:
+            print(f"\n{message}")
+
+conn.close()
```
Comment (dkijania) [inline check_db_genesis_balances.py:212]: It is always a risk closing connection like that as we can have some crash between. I don't have psycopg2 automaticallly closes connection unless you will access cursor like:
```
with psycopg2.connect(..) as conn:
    with conn.cursor() as curs:
       curs.execute(SQL)
```

**Review by dkijania — COMMENTED** [review]
Just placing check_gb_genesis_balances.py should be moved somewhere else. Other remarks are NITs :)

File: check_db_genesis_balances.py
Comment (joaosreis) [reply check_db_genesis_balances.py:1]: I wanted to commit it to `src/app/berkeley_migration` initially. Which of these paths should be the best?

**Review by dkijania — APPROVED** [review]

**Comment by joaosreis** [comment]
!ci-build-me

**Comment by joaosreis** [comment]
!ci-build-me

**Review by emberian — APPROVED** [review]
👍🏼 , will make sure on my end that the cache i add in #15310 doesn't interfere with this.

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**